File: modules/expo-terminal-keyboard/android/src/main/java/expo/modules/terminalkeyboard/ExpoTerminalKeyboardModule.kt
package expo.modules.terminalkeyboard

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoTerminalKeyboardModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoTerminalKeyboard")

    View(TerminalKeyboardView::class) {
      Events("onInput")

      Prop("focused") { view: TerminalKeyboardView, focused: Boolean ->
        view.setTerminalFocused(focused)
      }
    }
  }
}

File: modules/expo-terminal-keyboard/ios/ExpoTerminalKeyboardModule.swift
import ExpoModulesCore

public class ExpoTerminalKeyboardModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoTerminalKeyboard")

    View(TerminalKeyboardView.self) {
      Events("onInput")

      Prop("focused") { (view: TerminalKeyboardView, focused: Bool) in
        DispatchQueue.main.async {
          if focused {
            _ = view.becomeFirstResponder()
          } else {
            _ = view.resignFirstResponder()
          }
        }
      }
    }
  }
}
